utilities/pipelines/platform/Set-AvmGitHubIssueForWorkflow.ps1

```diff
@@ -0,0 +1,247 @@
+    #    Get all issues   #
+    #######################
+    $baseInputObject = @{
+        RepositoryOwner = $RepositoryOwner
+        RepositoryName  = $RepositoryName
+    }
+    if ($PersonalAccessToken) {
+        $baseInputObject['PersonalAccessToken'] = @{
+            PersonalAccessToken = $PersonalAccessToken
+        }
+    }
+
+    $issues = Get-GitHubIssueList @baseInputObject | Where-Object {
+        $_.title -match '^\[Failed pipeline\] .+'
+    }
+
+    ##########################
+    #    Get all workflows   #
+    ##########################
+
+    Write-Verbose 'Fetching current GitHub workflows' -Verbose
+    $workflows = Get-GitHubModuleWorkflowList @baseInputObject -Filter $PipelineFilter | Where-Object {
+        $_.name -notin $IgnoreWorkflows
+    }
+    Write-Verbose ('Fetched [{0}] workflows' -f $workflows.Count) -Verbose
+
+    ############################################
+    #   Fetching latest run of each workflow   #
+    ############################################
+    Write-Verbose '⏳ Fetching each workflow''s latest run using the main branch' -Verbose
+    $workflowRunsToProcess = [System.Collections.ArrayList]@()
+    $totalCount = $workflows.Count
+    $currentCount = 1
+    foreach ($workflow in $workflows) {
+
+        $percentageComplete = [math]::Round(($currentCount / $totalCount) * 100)
+        Write-Progress -Activity ('Fetching workflow [{0}]' -f $workflow.name) -Status "$percentageComplete% complete" -PercentComplete $percentageComplete
+        # Get relevant run
+        $latestWorkflowRun = Get-GitHubModuleWorkflowLatestRun @baseInputObject -WorkflowId $workflow.id
+
+        if ($latestWorkflowRun.status -eq 'completed') {
+            $workflowRunsToProcess += $latestWorkflowRun
+        }
+
+        $currentCount++
+    }
+
+    ############################
+    #   Processing workflows   #
+    ############################
+    $issuesCreated = 0
+    $issuesCommented = 0
+    $issuesClosed = 0
+    foreach ($workflowRun in $workflowRunsToProcess) {
+        $issueName = '[Failed pipeline] {0}' -f $workflowRun.name
+
+        if ($workflowRun.conclusion -eq 'failure') {
+            $failedRunText = 'Failed run: {0}' -f $workflowRun.url -replace 'api\.github.com\/repos', 'github.com'
+            $moduleName = $workflowRun.name.Replace('.', '/')
+
+            if ($issues.title -notcontains $issueName) {
+                # If no issue exists yet, create a new one
+                if ($PSCmdlet.ShouldProcess("Issue [$issueName]", 'Create')) {
+                    $issueUrl = gh issue create --title $issueName --body $failedRunText --label 'Type: AVM :a: :v: :m:,Type: Bug :bug:' --repo "$RepositoryOwner/$RepositoryName"
+                }
+                Write-Warning ('⚠️   Created issue {0} ({1}) as the module''s latest run in the main branch failed.' -f $issueUrl, $issueName)
+
+                $ProjectNumber = 538 # AVM - Issue Triage
+                $comment = @"
```

Comment:
why is the comment for orphaned created here? what about first checking if the module has an owner and assign either comment value via if-else? unless I'm missing something of course

Reply:
refactored it, but must re-test

Reply:
Retested successfully